staging/src/k8s.io/apiserver/pkg/server/egressselector/config.go

```diff
@@ -106,6 +110,22 @@ func validateDirectConnection(connection apiserver.Connection, fldPath *field.Pa
 	return nil
 }
 
+func validateUDSConnection(connection apiserver.Connection, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if connection.HTTPConnect != nil {
+		allErrs = append(allErrs, field.Invalid(
+			fldPath.Child("httpConnect"),
+			"direct",
+			"httpConnect config should be absent for direct connect"))
+	} else if connection.UDSName == "" {
```

Comment:
Is the "else if" necessary?
If connection has non-nil HTTPConnect and empty UDSName, we should return both error in allErrs, right?